pgmpy/identification/instrument.py

```diff
@@ -0,0 +1,148 @@
+from networkx.algorithms.dag import descendants
+
+from pgmpy.base import DAG
+from pgmpy.global_vars import logger
+from pgmpy.identification import BaseIdentification
+from pgmpy.inference.CausalInference import CausalInference
+
+
+class InstrumentalVariables(BaseIdentification):
+
+    def __init__(self, variant=None, scaling_indicators=None):
+        self.supported_graph_types = DAG
+        self.variant = variant
+        self.scaling_indicators = scaling_indicators
+
+    def _get_scaling_indicators(self, causal_graph):
+        latent_variables = causal_graph.get_role("latents")
+        observed_nodes = causal_graph.get_role("observed")
```

Comment:
Can we still use `nodes() - latents` as observed? We do not, by default, add the observed role to variables. We will either need to change this or assume that only latent roles are specified, and the rest are observed.